src/dfcx_scrapi/tools/test_cases_util.py

```diff
@@ -0,0 +1,704 @@
+            self._reverse_dict(dict_=commons_config["flows_map"]))
+        commons_config["intents_map_reverse"] = (
+            self._reverse_dict(dict_=commons_config["intents_map"]))
+        commons_config["pages_map_reverse"] = (
+            {f_id:self._reverse_dict(dict_=pages_map) \
+             for f_id, pages_map in dfcx_pages_map.items()})
+
+        return commons_config
+
+    def _get_new_test_config(
+        self,
+        source: dict,
+        target: dict,
+        test_case: types.TestCase) -> Union[types.TestConfig, None]:
+        """Update the test case types.test_config to the new types.test_config
+            Update the uuids of flow, page, intent by the display_name
+          args:
+            source: a dictionary of the source agent's flows, pages, and intents
+            target: a dictionary of the target agent's flows, pages, and intents
+            test_case: types.test_config object
+          returns:
+            types.test_config
+        """
+        source_flow_id = self._get_flow_id_from_test_config(test_case)
+        source_page_id = self._get_page_id_from_test_config(
+            test_case, source_flow_id)
+        source_flow = self._get_flow_display_name_by_id(
+            source_flow_id, source["flows_map"])
+        source_page = self._get_page_display_name_by_id(
+            source_flow_id, source_page_id, source["pages_map"])
+        target_test_config = types.TestConfig()
+
+        if test_case.test_config.flow:
+            target_flow_id = self._get_flow_id_by_flow_name(
+                flow_name=source_flow,
+                flows_map=target["flows_map_reverse"])
+            if target_flow_id:
+                target_test_config.flow = target_flow_id
+            else:
+                return None
+
+        elif test_case.test_config.page:
+            target_flow_id = self._get_flow_id_by_flow_name(
+                flow_name=source_flow,
+                flows_map=target["flows_map_reverse"])
+            target_page_id = self._get_page_id_by_page_name(
+                flow_id=target_flow_id,
+                page_name=source_page,
+                pages_map=target["pages_map_reverse"])
+            if target_page_id:
+                target_test_config.page = target_page_id
+            else:
+                return None
+
+        return target_test_config
+
+    def _get_flow_id_from_test_config(
+        self,
+        test_case: types.TestCase) -> str:
+        """Attempt to get the Flow ID from the Test Case Test Config."""
+
+        if "flow" in test_case.test_config:
+            return test_case.test_config.flow
+        elif "page" in test_case.test_config:
+            return "/".join(test_case.test_config.page.split("/")[:8])
+        else:
+            agent_id = "/".join(test_case.name.split("/")[:6])
+            return f"{agent_id}/flows/00000000-0000-0000-0000-000000000000"
+
+    def _get_page_id_from_test_config(
```

Comment:
@jkshj21 same as above. access from TestCases instead.

Reply:
Removed and it uses the dfcx_scrapi.core.test_cases class to call this function

Reply:
@kmaphoenix - hello, when i run pylint, I'm seeing errors below. It seems like I cannot access these functions because there is a "_" at the front, which I believe should only use them as an internal or protected function. 

```
test.py:219:12: W0212: Access to a protected member _get_flow_id_from_test_config of a client class (protected-access)
test.py:223:12: W0212: Access to a protected member _get_page_id_from_test_config of a client class (protected-access)

-----------------------------------
Your code has been rated at 9.92/10
```

Reply:
In that case, you can modify the method names in the TestCases class and make them non-private.

i.e.
`_get_flow_id_from_test_config` -> `get_flow_id_from_test_config`

Just make sure that when you're making the modification in your IDE you use the "rename all" setting or similar to rename all the references. once you've made the update, let me know and I can also run a check.

Reply:
@jkshj21 actually for now, let's just ignore the lint warnings and we can fix in another branch.

Just add the following to each line:
`# pylint: disable=W0212`